File: connectwave-frontend/src/app/services/like.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment.development';
import { Post } from '../models/post';

@Injectable({
  providedIn: 'root'
})
export class LikeService {

  constructor(private http: HttpClient) { }

  public saveLikeStatus(userId: number, postId: number, isLiked: string): Observable<Post> {
    return this.http.post<Post>(`${environment.hostUrl}/api/v1/likes/${userId}/${postId}/${isLiked}/saveLikeStatus`, {});
  }

  public getAllLikesByUser(userId: number): Observable<Post[]> {
    return this.http.get<Post[]>(`${environment.hostUrl}/api/v1/likes/${userId}/getAllLikesByUser`);
  }

  // public getNumLikesForPost(postId: number): Observable<number> {
  //   return this.http.get<number>(`${environment.hostUrl}/api/v1/likes/getNumLikesForPost/${postId}`);
  // }
}

File: connectwave-backend/src/main/java/com/abdulmo123/connectwave/service/LikeService.java
package com.abdulmo123.connectwave.service;

import com.abdulmo123.connectwave.model.entity.Post;
import com.abdulmo123.connectwave.model.response.HttpResponse;

import java.util.List;

public interface LikeService {

    HttpResponse getAllLikesByUser (Long userId);

    Post saveLikeStatus(Long userId, Long postId, String isLiked) throws Exception;

}
